File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        temp1=dict()
        temp2=dict()
        for x in nums1:
            temp1[x]=1 if x not in temp1 else temp1[x]+1
        for x in nums2:
            temp2[x]=1 if x not in temp2 else temp2[x]+1
        ans=[]
        for x in temp1.keys():
            if x in temp2:
                temp=min(temp1[x],temp2[x])
                for i in range(temp):
                    ans.append(x)
        return ans
        

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        List<Integer> store = new ArrayList<>();
        for(int n:nums1){
            if(mp.containsKey(n)){
                mp.put(n,mp.get(n)+1);
            }else{
                mp.put(n,1);
            }
        }
        for(int n:nums2){
            if(mp.containsKey(n)){
                store.add(n);
                mp.put(n,mp.get(n)-1);
                if(mp.get(n)==0){
                    mp.remove(n);
                }
            }
        }
        int []ans=new int[store.size()];
        for(int i=0;i<store.size();i++){
            ans[i]=store.get(i);
        }
        return ans;
    }
}
